tasks/src/main/scala/dagr/tasks/picard/CollectIlluminaLaneMetrics.scala

```diff
@@ -29,6 +29,23 @@ import dagr.tasks.DagrDef.DirPath
 
 import scala.collection.mutable.ListBuffer
 
+object CollectIlluminaLaneMetrics {
+  def convertBarcodeAndSkipToTemplate(readStructure: String): String = {
+    readStructure
```

Comment:
I find the `foldLeft` the most incomprehensible, and didn't find a way I liked to split it up.  It is what it is.